### PR TITLE
fedekunze/1550-1555 e2e voting and deposit

### LINKED ISSUE #1550: Add voting to E2E tests
Description:

Split from #1541 for ease of review

### LINKED ISSUE #1555: Add submit deposit to E2E tests
Description:

### PR DESCRIPTION
**Pull request by fedekunze**
Closes #1550
Closes #1555

_Description:_

Please review and merge #1671 first 

❤️ Thank you!

---

- [x] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [x] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: test/e2e/governance.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      await app.client
+        .$(`#cast-vote`)
+        .click()
+        .waitForVisible(
+          `//*[. = 'You have successfully voted Yes on proposal #1']`,
+          4 * 1000
+        ),
+      `successful vote`
+    )
+    // TODO: check if tally increased, blocked by SDK v.0.27
```
Comment (fedekunze) [inline test/e2e/governance.js]: ToDo in #1679

File: app/src/renderer/components/governance/PageProposal.vue [hunk 2]
```diff
@@ -216,8 +216,7 @@ export default {
     status() {
       if (this.proposal.proposal_status === `Passed`)
         return {
-          message: `This proposal has passed`,
-          color: `green`
+          message: `This proposal has passed`
```
[the remark this answers is not in the document]
Comment (jbibla) [reply app/src/renderer/components/governance/PageProposal.vue]: we're using the check icon ✅

File: test/e2e/delegation.js [hunk 3]
```diff
@@ -153,18 +161,44 @@ test(`delegation`, async function(t) {
...
+      ),
+      `Shows staking parameters`
+    )
+    await t.ok(
+      !(await app.client.isExisting(`//dd[contains(text(), "n/a")]`)),
+      `all parameters and pool fields are defined`
+    )
+    await t.ok(
+      !(await app.client.isExisting(`.tm-notification`, 4 * 1000)),
+      `should not get an notification error while fetching params and pool`
```
Comment (jbibla) [inline test/e2e/delegation.js]:
```suggestion
      `should not get a notification error while fetching params and pool`
```

File: test/e2e/governance.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+  t.test(`deposit`, async function(t) {
+    let deposit = () =>
+      $(`//dt[contains(text(), "Deposit")]`)
+        .$(`..`)
+        .$(`dd`)
+    let amount = parseInt((await deposit().getText()).split(` `)[0])
+    await t.ok(
+      await app.client.$(`.validator-profile__status.yellow`).isVisible(),
+      `the proposal is on deposit period`
```
Comment (jbibla) [inline test/e2e/governance.js]:
```suggestion
      `the proposal is open for deposits`
```

File: test/e2e/governance.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    )
+    await app.client.$(`#deposit-btn`).click()
+    await t.ok(
+      await app.client.$(`#modal-deposit`).isVisible(),
+      `opens modal deposit`
```
Comment (jbibla) [inline test/e2e/governance.js]:
```suggestion
      `opens deposit modal`
```

File: test/e2e/governance.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+          `//*[. = 'You have successfully deposited your steaks on proposal #1']`,
+          4 * 1000
+        ),
+      `successful deposit`
+    )
+    let newAmount = parseInt((await deposit().getText()).split(` `)[0])
+    t.equal(
+      newAmount,
+      amount + 10,
+      `increments the deposit displayed on the proposal page`
```
Comment (jbibla) [inline test/e2e/governance.js]:
```suggestion
      `increments the deposit count displayed on the proposal page`
```

File: test/e2e/governance.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    )
+    await closeNotifications(app)
+    t.end()
+  })
+
+  t.test(`vote`, async function(t) {
+    await t.ok(
+      await app.client.$(`.validator-profile__status.green`).isVisible(),
+      `the proposal is on voting period`
```
Comment (jbibla) [inline test/e2e/governance.js]:
```suggestion
      `the proposal is open for voting`
```

File: test/e2e/governance.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    )
+    await app.client.$(`#vote-btn`).click()
+    await t.ok(
+      await app.client.$(`#modal-vote`).isVisible(),
+      `opens modal vote`
```
Comment (jbibla) [inline test/e2e/governance.js]:
```suggestion
      `opens voting modal`
```

**Pull request merged**